### DISCUSSION TITLE
Proofreading

### DISCUSSION THREAD

**Discussion by ghost** [Jimskapt/rust-book-fr#269] [Q&A] [answered]
Hi! 👋
I haven't done any contribution yet (not at all), but I willing to proofread some chapters.

I may be mistaken, but I haven't seen a lot of informations about proofreading in both `CONTRIBUTING.md` and `/FRENCH/CONTRIBUTING.md`, the main content there is about translating, which is already done.

This may seem like a stupid question, but I'm not quite sure what exactly should I do while proofreading.
For now, I consider :
- checking the spelling and grammatical mistakes,
- checking that all makes sense so that it's understandable to everyone,

But should I also check that the content of the chapter isn't outdated compared to the last version of the English book ?
And should I send a message on the associated PR, to say e.g. "hey guys, i'm currently proofreading ch10-00" ?

And if I've missed anything, please tell me, I'm a cool guy 😄 !

Thanks for the answer !

And btw, i'm learning Rust in the meantime, but the chapters I'll proofread are only the ones that I already read in English.

**Comment by Jimskapt** [comment, marked as answer]
Hello Martin, and welcome here ! 😊

There is no stupid questions, only topics that need to be documented 😁

1. The main reason we need proofreading is that we need to find translation mistakes that will cause readers to learn something wrong about Rust.
2. The secondary reason is to rework wording in order to make this book easier to read _(but it will still ensures accuracy with English content)_.

We occasionally update outdated translated content, but only content already proofreaded and merged in `french-release` branch.
For not proofreaded content like currently `ch15-02` for example (on `FR-ch15-02`), we should not update it when proofreading it. So the proofreading will be do between commented English content and translated content, on the same markdown page.
We have to wait the positive review of the proofreading to update content.
When proofreeding accepted, we will then do the update of the outdated content and merge its branch on `french-release` branch.

You can find my recommended process of proofreading in #249.
You can also take a look on [closed proofreading PR](https://github.com/Jimskapt/rust-book-fr/pulls?q=is%3Apr+is%3Aclosed), especially the ones done by @joeljpresent 😉

If you want to start contributing now, I suggest you to start small and keep waiting some review of your first PR before doing your work in other parts of the book.

I hope I've responded to your questions, please feel free to ask back here if you need ! 😊
